docs/circuit-proof-math.md

```diff
@@ -1,90 +1,188 @@
 Arithmetic Circuit Proofs
 =========================
 
```

Comment:
Do we want to describe this as "Arithmetic Circuit Proofs" or as "Rank-1 Constraint System Proofs"?  We started out using "circuit" everywhere, but what we've actually implemented is a constraint system, and we did it differently than the paper. So maybe we should change the notes to reflect that?

Reply:
I think it would make sense to start out with "this is an explanation of R1CS" and then have a section describing "what is R1CS", and then after that a section on "how we go from R1CS to circuit", and "what is an arithmetic circuit". 

Right now, it's sort of confusing because the doc is about "arithmetic circuit proofs" but it is not super clear on how constraints and R1CS come into the picture / why we care about circuit->R1CS conversion.

Reply:
In the follow-up commits I've removed all mentions of "circuits" entirely. I did not go outside the notes into source code with more renaming as it's out of scope for this PR.